stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt

```diff
@@ -1328,21 +1329,16 @@ internal class ChannelControllerImpl(
         newerMessagesOffset: Int,
         olderMessagesOffset: Int
     ): Result<Message> {
-        val result = client.getMessage(messageId).execute()
-        val message = if (result.isSuccess) {
-            result.data()
-        } else {
-            null
-        }
-        return if (message != null) {
-            _messages.value = emptyMap()
-            upsertMessage(message)
-            loadNewerMessages(newerMessagesOffset)
-            loadOlderMessages(olderMessagesOffset)
-            Result(message)
-        } else {
-            Result(null, ChatError("Error while fetching message from backend. Message id: $messageId"))
+        val result = withContext(domainImpl.scope.coroutineContext) { client.getMessage(messageId).execute() }
```

Comment:
We're already in this same context when we're running this method, so `withContext` doesn't do anything. We can freely block the thread with `execute()`, or go to yet another thread with `await()` to be super sure that whoever calls this won't be blocked.